pyvista/plotting/axes_actor.py

```diff
@@ -1,176 +1,416 @@
 
     @visibility.setter
     def visibility(self, value: bool):  # numpydoc ignore=GL08
-        return self.SetVisibility(value)
+        self.SetVisibility(value)
 
     @property
     def total_length(self) -> tuple:  # numpydoc ignore=RT01
-        """Return or set the length of all axes.
+        """Total length of each axis (shaft plus tip).
 
         Examples
         --------
         >>> import pyvista as pv
-        >>> axes = pv.Axes()
-        >>> axes.axes_actor.total_length
+        >>> axes_actor = pv.AxesActor()
+        >>> axes_actor.total_length
         (1.0, 1.0, 1.0)
-        >>> axes.axes_actor.total_length = 1.2
-        >>> axes.axes_actor.total_length
+        >>> axes_actor.total_length = 1.2
+        >>> axes_actor.total_length
         (1.2, 1.2, 1.2)
-        >>> axes.axes_actor.total_length = (1.0, 0.9, 0.5)
-        >>> axes.axes_actor.total_length
+        >>> axes_actor.total_length = (1.0, 0.9, 0.5)
+        >>> axes_actor.total_length
         (1.0, 0.9, 0.5)
 
         """
         return self.GetTotalLength()
 
     @total_length.setter
     def total_length(self, length):  # numpydoc ignore=GL08
-        if isinstance(length, Iterable):
+        if isinstance(length, Sequence):
             self.SetTotalLength(length[0], length[1], length[2])
         else:
             self.SetTotalLength(length, length, length)
 
     @property
     def shaft_length(self) -> tuple:  # numpydoc ignore=RT01
-        """Return or set the length of the axes shaft.
+        """Normalized length of the shaft for each axis.
+
+        Values should be in range ``[0, 1]``.
 
         Examples
         --------
         >>> import pyvista as pv
-        >>> axes = pv.Axes()
-        >>> axes.axes_actor.shaft_length
+        >>> axes_actor = pv.AxesActor()
+        >>> axes_actor.shaft_length
         (0.8, 0.8, 0.8)
-        >>> axes.axes_actor.shaft_length = 0.7
-        >>> axes.axes_actor.shaft_length
+        >>> axes_actor.shaft_length = 0.7
+        >>> axes_actor.shaft_length
         (0.7, 0.7, 0.7)
-        >>> axes.axes_actor.shaft_length = (1.0, 0.9, 0.5)
-        >>> axes.axes_actor.shaft_length
+        >>> axes_actor.shaft_length = (1.0, 0.9, 0.5)
+        >>> axes_actor.shaft_length
         (1.0, 0.9, 0.5)
 
         """
         return self.GetNormalizedShaftLength()
 
     @shaft_length.setter
-    def shaft_length(self, length):  # numpydoc ignore=GL08
-        if isinstance(length, Iterable):
+    def shaft_length(self, length: Union[float, Sequence[float]]):  # numpydoc ignore=GL08
+        if isinstance(length, Sequence):
             self.SetNormalizedShaftLength(length[0], length[1], length[2])
         else:
             self.SetNormalizedShaftLength(length, length, length)
 
     @property
-    def tip_length(self) -> tuple:  # numpydoc ignore=RT01
-        """Return or set the length of the tip.
+    def tip_length(self) -> Tuple[float, float, float]:  # numpydoc ignore=RT01
```

Comment:
We have the `Vector` type for `Tuple[float, float, float]`. See:

https://github.com/pyvista/pyvista/blob/76aa761dac8dddf9995a49872399e4f9a406ecf8/pyvista/core/_typing_core.py#L8

Reply:
This is the getter, which will definitely return a `tuple`. Do you want it as `Vector` anyway?

The setter looks like its type can be changed to use `Vector` if you want though (along with all other variables and the class docstring).

Reply:
Ah, good catch. If the returned value is always a tuple, then lest keep it as is.

Reply:
>The setter looks like its type can be changed to use Vector if you want though

The setter should probably use the `Vector` type

Reply:
The typing for the whole class was updated in 769930c9b98676ae02e83839ff37fe052b1edf8d